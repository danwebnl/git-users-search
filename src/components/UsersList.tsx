import * as React from 'react';
import { Link } from 'react-router-dom';

import { ListContainer } from './styles';

interface IProps {
    users: [{
        login: string
    }];
}

const UsersList = (props: IProps) => (
    <ListContainer>
        <h2>Search Results</h2>
        <ul>
            {
                props.users.map((user: { login: string }, index: number) => (
                    <li key={`user${index}`}>
                        <Link to={`/user/${user.login}`}>{user.login}</Link>
                    </li>
                ))
            }
        </ul>
    </ListContainer>
);

export default UsersList;
