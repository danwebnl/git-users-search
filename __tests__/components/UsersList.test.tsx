import * as React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import UsersList from '../../src/components/UsersList';

const testUsers: any = [{ login: 'acount-a' }, { login: 'acount-b' }];
const testUsersUpdate: any = [{ login: 'acount-a' }, { login: 'acount-c' }];

describe('<UsersList /> ', () => {

    it('renders', () => {
        shallow(<UsersList users={testUsers} />);
    });

    it('matches the snapshot', () => {
        const wrapper = shallow(<UsersList users={testUsers} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('updates via props', () => {
        const wrapper = shallow(<UsersList users={testUsersUpdate} />);
        expect(toJSON(wrapper)).toMatchSnapshot();

        wrapper.setProps(testUsers);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});
