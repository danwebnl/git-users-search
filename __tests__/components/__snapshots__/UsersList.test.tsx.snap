// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersList />  matches the snapshot 1`] = `
<styled.div>
  <h2>
    Search Results
  </h2>
  <ul>
    <li
      key="user0"
    >
      <Link
        replace={false}
        to="/user/acount-a"
      >
        acount-a
      </Link>
    </li>
    <li
      key="user1"
    >
      <Link
        replace={false}
        to="/user/acount-b"
      >
        acount-b
      </Link>
    </li>
  </ul>
</styled.div>
`;

exports[`<UsersList />  updates via props 1`] = `
<styled.div>
  <h2>
    Search Results
  </h2>
  <ul>
    <li
      key="user0"
    >
      <Link
        replace={false}
        to="/user/acount-a"
      >
        acount-a
      </Link>
    </li>
    <li
      key="user1"
    >
      <Link
        replace={false}
        to="/user/acount-c"
      >
        acount-c
      </Link>
    </li>
  </ul>
</styled.div>
`;

exports[`<UsersList />  updates via props 2`] = `
<styled.div>
  <h2>
    Search Results
  </h2>
  <ul>
    <li
      key="user0"
    >
      <Link
        replace={false}
        to="/user/acount-a"
      >
        acount-a
      </Link>
    </li>
    <li
      key="user1"
    >
      <Link
        replace={false}
        to="/user/acount-c"
      >
        acount-c
      </Link>
    </li>
  </ul>
</styled.div>
`;
